clickhouse-client/src/main/java/com/clickhouse/client/config/ClickHouseClientOption.java

```diff
@@ -252,15 +252,59 @@ public enum ClickHouseClientOption implements ClickHouseOption {
      * Socket timeout in milliseconds.
      */
     SOCKET_TIMEOUT("socket_timeout", 30 * 1000, "Socket timeout in milliseconds."),
+    /**
+     * Whether allows for the reuse of local addresses and ports. See
+     * {@link java.net.StandardSocketOptions#SO_REUSEADDR}.
+     */
+    SOCKET_REUSEADDR("socket_reuseaddr", false,
+            "Whether allows for the reuse of local addresses and ports. "
+                    + "Only works for client using custom Socket(e.g. TCP client or HTTP provider with custom SocketFactory etc.)."),
```

Comment:
@zhicwu does custom SocketFactory refers to `ApacheHttpConnectionImpl.SocketFactory` and `ApacheHttpConnectionImpl.SSLSocketFactory` ?

I am wondering if it is possible to inject a custom socket factory.

Reply:
> Does custom SocketFactory refers to ApacheHttpConnectionImpl.SocketFactory and ApacheHttpConnectionImpl.SSLSocketFactory?

Yes, for now. Ideally we should have a custom socket factory for not only Apache HTTP Client but also others(i.e. NIO-based implementation).

> I am wondering if it is possible to inject a custom socket factory.

Sorry it's not supported at this point. What's the purpose of injecting custom socket factory? Is it to use consistent socket options across datasources?

Reply:
Thanks for confirming. Yes, I would like to create a custom socket.

Reply:
Are you planning to include that feature in the future?

Reply:
Thanks for confirming. Yes, I would like to create a custom socket.

Are you planning to include that feature in the future? Other JDBC drivers(eg. mysql) offer such capabilities.

Reply:
Thanks for the suggestion. Just created a separate issue for this, should not be hard to add the feature.